### DISCUSSION TITLE
Access parent record on associations actions

### DISCUSSION THREAD

**Discussion by coatezy** [avo-hq/avo#3159] [Help] [answered]
Hi all,

I am currently evaluating Avo with the plan to use Advanced. I've hit a blocker with the community edition that I was hoping could be overcome, either using the community plan or the pro/advanced.

Instead of attaching a course to a group I'd like to deep clone a course template and attach it to a course. I couldn't see a graceful way to modify the default "Attach course" action so I reached for a custom action, it looks like this could allow me to achieve my requirements if I could pass the group id to the action so I can attach the cloned course to the group.

If this possible or is there an inherent limitation that cannot be overcome right now?

Thanks
Tom

<img width="1242" alt="CleanShot 2024-08-20 at 21 54 58@2x" src="https://github.com/user-attachments/assets/d32ca773-55d0-4c85-8213-1a738aacea75">

**Comment by Paul-Bob** [comment, marked as answer]
Hello @coatezy 

> I am currently evaluating Avo with the plan to use Advanced. I've hit a blocker with the community edition that I was hoping could be overcome, either using the community plan or the pro/advanced.

This specific blocker can be overcome by the community plan. On the `advanced` one you could extract the action from the list by using the [customizable controls feature](https://docs.avohq.io/3.0/customizable-controls.html).

>  I couldn't see a graceful way to modify the default "Attach course" action so I reached for a custom action

The custom actions fit perfectly for this use case.

> it looks like this could allow me to achieve my requirements if I could pass the group id to the action so I can attach the cloned course to the group.
If this possible or is there an inherent limitation that cannot be overcome right now?

I've explored this, and while the parent record ID (group ID) might seem inaccessible at first glance, it is possible to pass it as an argument.

Here’s how you can achieve this, in your `Course` resource, where you define the action:
```ruby
 def actions
    action Avo::Actions::ActionName, arguments: -> do
      if @record.class == Group # or @record.is_a?(Group)
        {
          parent_param: @record.to_param
        }
      else
        {

        }
      end
    end
  end
```

Now in `Avo::Actions::ActionName` on `def fields` or `def handle` methods, you should be able to access it by `arguments[:parent_param]`.

Let me know if you find any obstacles while implementing this.

**Reply by Paul-Bob** [reply]
I created [this issue](https://github.com/avo-hq/avo/issues/3165) that tracks the parent record access on child's actions

**Comment by coatezy** [comment]
This worked perfectly, thanks @Paul-Bob! Hope this discussion and the ticket you created helps others too.

**Reply by Paul-Bob** [reply]
Cool @coatezy, anytime!

I changed the title to help others locate it more easily if they face a similar problem.
